File: 0693/693.py
class Solution:
    def binary_number_with_alternating_bits(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.binary_number_with_alternating_bits(root.left), self.binary_number_with_alternating_bits(root.right))

File: 0693/693.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var binaryNumberWithAlternatingBit = function(root) {
    if (!root) return 0;
    return 1 + Math.max(binaryNumberWithAlternatingBit(root.left), binaryNumberWithAlternatingBit(root.right));
};